File: Code/containsDuplicate_q217.py
"""
    Given an integer array nums, return true if any value appears at least twice in the array, 
    and return false if every element is distinct.
"""


class Solution(object):
    def containsDuplicate(self, nums):
        seen = set()
        has_dulplicates = False

        for num in nums:
            if num in seen:
                has_dulplicates = True
                return has_dulplicates
            seen.add(num)

        return has_dulplicates


solution = Solution()
print(solution.containsDuplicate([1, 0, 1]))

File: Code/containsDuplicate_q217.js
/* 
    Given an integer array nums, return true if any value appears at least twice in the array, 
    and return false if every element is distinct.
*/

function containsDuplicate(nums) {
    let seen = new Set()

    for (i = 0; i < nums.length; i++) {
        if (seen.has(nums[i])) {
            return true
        }
        seen.add(nums[i])
    }

    return false
}

console.log(containsDuplicate([1, 2, 3, 1]))
console.log(containsDuplicate([1, 2, 3, 4]))
